audobject/core/api.py

```diff
@@ -0,0 +1,108 @@
+import os
+import typing
+
+import oyaml as yaml
+
+from audobject.core.object import Object
+import audobject.core.utils as utils
+
+
+def _decode_value(
+        value_to_decode: typing.Any,
+        **kwargs,
+) -> typing.Any:
+    r"""Decode value."""
+    if value_to_decode:  # not empty
+        if isinstance(value_to_decode, list):
+            return [
+                _decode_value(v, **kwargs) for v in value_to_decode
+            ]
+        elif isinstance(value_to_decode, dict):
+            name = next(iter(value_to_decode))
+            if isinstance(name, Object) or utils.is_class(name):
+                return load_from_dict(value_to_decode, **kwargs)
+            else:
+                return {
+                    k: _decode_value(v, **kwargs) for k, v in
+                    value_to_decode.items()
+                }
+    return value_to_decode
+
+
+def load_from_dict(
+        d: typing.Dict[str, typing.Any],
+        root: str = None,
+        **kwargs,
+) -> 'Object':
+    r"""Create object from dictionary.
+
+    Args:
+        d: dictionary with variables
+        root: if dictionary was read from a file, set to source directory
+        kwargs: additional variables
```

Comment:
What are additional variables?

Reply:
That's indeed not very precise, I rephrased to:

![image](https://user-images.githubusercontent.com/10383417/135042855-972cfa8b-f698-4c6f-bef0-7e9e04ba0403.png)